src/commands/executeFunction/executeEventGridFunction.ts

```diff
@@ -0,0 +1,156 @@
+import { executeFunctionWithInput } from './executeFunction';
+
+type FileMetadata = {
+    name: string;
+    path: string;
+    sha: string;
+    size: number;
+    url: string;
+    html_url: string;
+    git_url: string;
+    download_url: string;
+    type: string;
+    _links: {
+        self: string;
+        git: string;
+        html: string;
+    };
+};
+
+const sampleFilesUrl =
+    'https://api.github.com/repos/Azure/azure-rest-api-specs/contents/specification/eventgrid/data-plane/' +
+    '{eventSource}' +
+    '/stable/2018-01-01/examples/cloud-events-schema/';
+
+export async function executeEventGridFunction(context: IActionContext, node: FunctionTreeItemBase | IFunction): Promise<void> {
+    // Prompt for event source
+    const eventGridSourcePicks: IAzureQuickPickItem<EventGridSource | undefined>[] = supportedEventGridSources.map((source: EventGridSource) => {
+        return {
+            label: supportedEventGridSourceLabels.get(source) || source,
+            data: source,
+        };
+    });
+    const eventSource: EventGridSource =
+        (
+            await context.ui.showQuickPick(eventGridSourcePicks, {
+                placeHolder: localize('selectEventSource', 'Select the event source'),
+                stepName: 'eventGridSource',
+            })
+        ).data ?? 'Microsoft.Storage';
+
+    // Get sample files for event source
+    const samplesUrl = sampleFilesUrl.replace('{eventSource}', eventSource);
+    const sampleFiles: FileMetadata[] = await feedUtils.getJsonFeed(context, samplesUrl);
+    const fileNames: string[] = sampleFiles.map((fileMetadata) => fileMetadata.name);
+
+    // Prompt for event type
+    const eventTypePicks: IAzureQuickPickItem<string | undefined>[] = fileNames.map((name: string) => ({
+        data: name,
+        // give human-readable name for event type from file name
+        label: name
+            .replace(/\.json$/, '')
+            .split('_')
+            .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
+            .join(' '),
+    }));
+    const selectedFileName: string =
+        (
+            await context.ui.showQuickPick(eventTypePicks, {
+                placeHolder: localize('selectEventType', 'Select the event type'),
+                stepName: 'eventType',
+            })
+        ).data ?? 'blob_created.json';
+
+    // Get selected contents of sample request
+    const selectedFileUrl = sampleFiles.find((fileMetadata) => fileMetadata.name === selectedFileName)?.download_url || sampleFiles[0].download_url;
+    const selectedFileContents: {} = await feedUtils.getJsonFeed(context, selectedFileUrl);
+
+    // Prompt for whether to send or modify the sample request
+    const shouldModify: boolean =
+        (
+            await context.ui.showQuickPick(
+                [
+                    {
+                        label: localize('sendSample', 'Send sample request'),
```

Comment:
@alexweininger brought it up offline, and I was thinking something kind of similar: are there many scenarios where the user would want to just send the sample request without editing it? It seems like there are typically a lot of properties that need to be edited for it to actually emulate an Event Grid response.

I'm kind of leaning towards just skipping this prompt and always spit them out to the modify request.

Reply:
> I'm kind of leaning towards just skipping this prompt and always spit them out to the modify request.

I'm leaning towards this as well

Reply:
Removed.